components/aws/athena/query/src/query.py

```diff
@@ -24,16 +24,29 @@ def get_client(region=None):
   client = boto3.client('athena', region_name=region)
   return client
 
-def query(client, query, database, output):
-  response = client.start_query_execution(
-      QueryString=query,
-      QueryExecutionContext={
-          'Database': database
-          },
-      ResultConfiguration={
-          'OutputLocation': output,
-          }
-      )
+def query(client, query, database, output, workgroup=None):
+
+  if workgroup:
```

Comment:
Is there more elegant way to handle optional arguments here? If we have extra configs, we will have lots of if else and nested if else logic here.

Reply:
Yeah, sorry about that. The camelcase args aree not super elegant either (boto3). 

I can change it to: 

```python
arg_dict = dict(QueryString=query,
        QueryExecutionContext={
            'Database': database
            },
        ResultConfiguration={
            'OutputLocation': output,
            }
        )
    if workgroup:
        arg_dict.update(dict(WorkGroup=workgroup))

    response = client.start_query_execution(**arg_dict)
``` 

Unless you have a better option?

Reply:
I accidentally added black formatting in the latest commit. Sorry in advance. if the changes are not as clear. But it's essentially the above. Except `arg_dict` is renamed to `params`